### PR TITLE
`ApplicationState` Enhancements - Documentation

### PR DESCRIPTION
**Pull request by majedelass**
### :pushpin: Summary

If merged, this PR will update the documentation for the `ApplicationState` component.

### :hammer_and_wrench: Detailed description

Outlining all the enhancements coming to the component such as `alignment`, `media` and a more flexible footer with additional CTA types (like `button` and `dropdown`).

### :link: External links

Jira ticket: [HDS-3543](https://hashicorp.atlassian.net/browse/HDS-3543)
Figma file: [Link](https://www.figma.com/design/noyY6dUMDYjmySpHcMjhkN/branch/PHekpm994tyTcf1fIY2Pgn/HDS-Components?m=auto&t=ZZbO5IXZr4lxXZwb-1)

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3543]: https://hashicorp.atlassian.net/browse/HDS-3543?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by MelSumner — APPROVED** [review]

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (zamoore) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:37]: This works by passing an icon name to the `Header` subcomponent.

For example:

```
<ApplicationState as |A|>
  <A.Header @icon="alert-circle" @title="Danger!" />
</ApplicationState>
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (zamoore) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:53]: Maybe a callout for the errorCode attribute? Not sure if we prefer people to use it where applicable, or if it's just an option.

**Review by zamoore — COMMENTED** [review]
Looking really good so far!

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (majedelass) [reply website/docs/components/application-state/partials/guidelines/guidelines.md:37]: Maybe this should be reworded to "when set" instead of "when set to true."

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (majedelass) [reply website/docs/components/application-state/partials/guidelines/guidelines.md:59]: The errorCode was mentioned in the section above this one. This is specifically talking about adding more details in the body if this application state is associated to an error.

File: website/docs/components/application-state/index.md
Comment (majedelass) [inline website/docs/components/application-state/index.md:1]: @didoo @zamoore I believe this needs to have "Version History" section associated to it. You might need to sync on what content needs to be in there.

File: packages/components/src/components/hds/application-state/footer.hbs
Comment (majedelass) [inline packages/components/src/components/hds/application-state/footer.hbs:13]: @zamoore Was this update meant to be done in this PR vs [this](https://github.com/hashicorp/design-system/pull/2274)?

Comment (didoo) [reply packages/components/src/components/hds/application-state/footer.hbs:13]: Wait, are we allowing to use a dropdown in the footer of the application state? 🤔

File: website/docs/components/application-state/index.md
Comment (didoo) [reply website/docs/components/application-state/index.md:1]: I'll take care of it, once the component and the docs are ready to be merged

File: packages/components/src/components/hds/application-state/footer.hbs
Comment (majedelass) [reply packages/components/src/components/hds/application-state/footer.hbs:13]: Yes - l[ink to comment.](https://github.com/hashicorp/design-system/pull/2274/files#r1693318678)

Comment (didoo) [reply packages/components/src/components/hds/application-state/footer.hbs:13]: @majedelass see https://github.com/hashicorp/design-system/pull/2274/files#r1696874211

File: website/docs/components/application-state/partials/code/how-to-use.md
Comment (KristinLBradley) [inline website/docs/components/application-state/partials/code/how-to-use.md:78]: Could you please add an alt attribute to this `img` tag example?

(All img tags should include an alt attribute. If the image is decorative only then you can leave the alt value empty.)

Examples:
```
<img src="helios-logo.png" alt="Helios" />

<img src="decoration.png" alt="" />
```

Comment (KristinLBradley) [reply website/docs/components/application-state/partials/code/how-to-use.md:78]: Please add alt attributes to below img tags also.

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (KristinLBradley) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:59]: This part of the sentence seems awkward/unclear to me: "creating this new objective"

Do you mean provide info to the user on how/why they can add content?

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (KristinLBradley) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:69]:
```suggestion
![A button set showing a dropdown, secondary button, and stand alone link](/assets/components/application-state/application-state-footer-action-types.png)
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (KristinLBradley) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:73]: This part seems unclear to me "used to emphasize the importance of the action in the Application State with visual hierarchy".

Isn't the main purpose of Buttons & Links to provide the user with actionable options?

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (KristinLBradley) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:77]: This phrasing seems awkward and unclear: "do not display two exact actions at the same time in different areas of the UI"

File: packages/components/src/components/hds/application-state/footer.hbs
Comment (zamoore) [reply packages/components/src/components/hds/application-state/footer.hbs:13]: Yes, let me move this over. 🤦

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (jorytindall) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:6]: [suggestion]
```suggestion
- When emphasis is needed on the creation of a new object within a null state.
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (jorytindall) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:19]: [question] Is this also true depending on the container the component is used within?

```suggestion
By default, the Application State has horizontal auto margins applied to it, always centering it on the page or containing element. This can be overridden with CSS properties.
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (jorytindall) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:23]: [suggestion] These could be headlines, but I'm not opinionated about it. Just noting it in case it might be useful to link to them directly.
```suggestion
### Center alignment
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (jorytindall) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:27]:
```suggestion
### Left alignment
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (jorytindall) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:34]: [nit] If this is going to be the start of a new paragraph, there needs to be an extra return in between lines 33 and 34. Otherwise the content is combined in the same paragraph, which for consistency should probably be reflected in markdown.

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (jorytindall) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:36]: [nit] Vault Secrets is a proper noun.

```suggestion
![Empty state for Vault Secrets, guiding user to create new secrets or importing them](/assets/components/application-state/application-state-media-slot-spot-illustration-center-alignment.png)
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (jorytindall) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:55]: I'm not 100% clear what you mean by "available", is this referring to if an application runs into an error that corresponds with a specific code?

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (jorytindall) [reply website/docs/components/application-state/partials/guidelines/guidelines.md:59]: Did you mean this?

```suggestion
Focus on relevant information and avoid unnecessary details. If there is an error, include suggestions or guidance for how the user can resolve the issue, if possible. If no objects are found (zero/empty state), provide a brief explanation on how creating this new object will benefit the user. 
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (jorytindall) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:67]: [suggestion] This might read a bit cleaner in the last sentence.

```suggestion
The Application State supports up to three actions, including [Dropdown](/components/dropdown), [Standalone Link](/components/link/standalone) and [Button](/components/button). Use footer actions to help users resolve errors or access issues with actionable steps.
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (jorytindall) [reply website/docs/components/application-state/partials/guidelines/guidelines.md:77]: Maybe try this?

```suggestion
When there is an empty state that occupies the majority of the page, do not display two similar actions in different areas of the UI. In this example, there is a primary button in the Page Header and in the Application State. 
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (jorytindall) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:113]: I'm not actually sure on this, would benefit from a second opinion, but saying "are used" reads weird to me because they aren't necessarily an intentional use, more of a fallback for when there isn't anything.

```suggestion
An empty state occurs when a user has yet to create an object. Illustrations are placed using the `media` slot to further elevate the experience and express the purpose of the object.
```

File: website/docs/components/application-state/partials/guidelines/overview.md
Comment (jorytindall) [inline website/docs/components/application-state/partials/guidelines/overview.md:1]: [nit] Oxford comma?

```suggestion
The Application State is used to communicate status, errors, and zero/empty states.
```

**Review by jorytindall — COMMENTED** [review]
Couple of pretty basic comments, but this is looking solid!

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (majedelass) [reply website/docs/components/application-state/partials/guidelines/guidelines.md:113]: Agreed, updating.

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (majedelass) [reply website/docs/components/application-state/partials/guidelines/guidelines.md:34]: Yeah, ill update it to this:
```suggestion

This provides additional visual prominence while also elevating the brand experience. If the illustration has a circular container, we recommend setting the `alignment` to `center`.
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (KristinLBradley) [reply website/docs/components/application-state/partials/guidelines/guidelines.md:74]: Thanks. I would look to explain why someone would want to use a button vs. a link & vice versa and how that impacts visual hierarchy.

(Btw, would we also want to mention appropriate usage apart from visual designs of HTML links/anchor tags vs. HTML buttons or is that outside the scope here?)

Comment (MelSumner) [reply website/docs/components/application-state/partials/guidelines/guidelines.md:74]: +1 to what @KristinLBradley has said here. We have already taught folks that a button is for an action (submitting a form, opening a modal, etc), and a link is for a URL. 

We are already stretching things by having links that visually present as buttons (and/or vice-versa). I wonder if there's some way to indicate this in the design guidance?

I still want us to be encouraging folks to do the correct thing for a11y, KWIM?

File: website/public/assets/components/application-state/application-state-anatomy-page.png
Comment (didoo) [inline website/public/assets/components/application-state/application-state-anatomy-page.png:1]: @majedelass this new image still contain a reference to the `separator`

File: website/docs/components/application-state/partials/specifications/anatomy.md
Comment (didoo) [inline website/docs/components/application-state/partials/specifications/anatomy.md:12]: This should be removed
```suggestion
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (didoo) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:27]: Given that this is the default, should this come first?

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (didoo) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:15]:
```suggestion
The Application State supports two alignment options: `left` (default) and `center`. The alignment affects text alignment, action placement/alignment in the footer, and media placement; however, it does not change the default page alignment.
```

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (didoo) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:33]: In theory not just illustrations. Should we be more generic?

**Review by didoo — COMMENTED** [review]
Left a few comments, nothing major.
Let me know when fixed so I can approve.

File: website/docs/components/application-state/partials/specifications/anatomy.md
Comment (majedelass) [reply website/docs/components/application-state/partials/specifications/anatomy.md:12]: Good catch 👍

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (majedelass) [reply website/docs/components/application-state/partials/guidelines/guidelines.md:33]: Correct, however from a UX perspective this is our recommendation which is why it's written as such. We're trying to drive for more consistency.

**Review by didoo — APPROVED** [review]
Approving so it won't block the release

File: website/docs/components/application-state/partials/guidelines/guidelines.md
Comment (majedelass) [inline website/docs/components/application-state/partials/guidelines/guidelines.md:76]: @KristinLBradley @MelSumner is this a better approach to this section? Tried to be a little more specific when to use what without rewriting each component's docs.

**Review by KristinLBradley — APPROVED** [review]

**Review by jorytindall — APPROVED** [review]
:shipit:

**Pull request merged**
